R/zchunk_L125.bcoc_unmgd_R_S_T_Y.R

```diff
@@ -35,77 +34,191 @@ module_emissions_L125.bcoc_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
+
+    # RCP emissions by GCAM region
+    # Add em species identifier, map to GCAM region, convert from kg to Tg,
+    # aggregate to GCAM region
+    RCP_BC_2000 <- RCP_BC_2000 %>% mutate(Non.CO2 = "BC")
+    RCP_OC_2000 <- RCP_OC_2000 %>% mutate(Non.CO2 = "OC")
+    L125.RCP <- bind_rows( RCP_BC_2000, RCP_OC_2000 ) %>%
+      left_join_error_no_match(iso_GCAM_regID %>% select(iso, GCAM_region_ID), by = "iso") %>%
+      select( GCAM_region_ID, Non.CO2, lcf, sav) %>%
+      mutate(lcf = lcf*CONV_KG_TO_TG) %>% # convert from kg to Tg
+      mutate(sav = sav*CONV_KG_TO_TG) %>% # convert from kg to Tg
+      group_by(GCAM_region_ID, Non.CO2) %>%
+      summarize_if(is.numeric, sum) # sum by GCAM region
+
+    # Compute grassland emissions factors by GCAM region
+    # Because grassland and forest fire emissions scale with land quantity, the coefs can be computed at the regional level
+    # and applied equally to all land use regions (GLUs) thereafter.
+    # (However deforestation emissions are assigned from the region
+    # to the GLUs according to relative shares of deforested land, different from forest land cover.)
+
+    L125.bcoc_tgbkm2_R_grass_2000 <- L124.LC_bm2_R_Grass_Yh_GLU_adj %>%
+      filter(year == 2000) %>%
+      group_by(GCAM_region_ID, Land_Type, year) %>%
+      summarize_at(vars(value),sum) %>% # aggregate grassland land area by regions/land type
+      repeat_add_columns(tibble::tibble(Non.CO2 = unique( L125.RCP$Non.CO2 ))) %>% # repeat for both BC and OC
+      left_join_error_no_match(L125.RCP %>% select(-lcf), by = c("GCAM_region_ID", "Non.CO2")) %>% # add emissions to land region area
+      mutate(em_factor = sav/value) %>% # calculate emission factor (emissions/area)
+      select( GCAM_region_ID, Land_Type, Non.CO2, em_factor ) %>%
+      arrange(Non.CO2, Land_Type,GCAM_region_ID)
+
+    # Compute forest fire emissions factors by GCAM region
+    # aggregate grassland area by region/land type
+    L125.bcoc_tgbkm2_R_forestfire_2000 <- L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj %>%
+      filter(year == 2000) %>%
+      select(-year) %>%
+      group_by(GCAM_region_ID, Land_Type) %>%
+      summarize_at(vars(value),sum) %>% # aggregate grassland area to regions/land type
+      dplyr::rename(FF_driver = value)
+  # select only deforestation coefficient years (used to estiamte rate of change - D-driver)
+    L125.bcoc_tgbkm2_R_GLU_defor_2000 <- L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj %>%
+      filter( year %in% emissions.DEFOREST_COEF_YEARS) %>%
+      group_by(GCAM_region_ID, Land_Type, GLU) %>%
+      spread(year, value) %>%
+      mutate(D_driver = pmax(`2000`-`2005`,0) / (emissions.DEFOREST_COEF_YEARS[2] - emissions.DEFOREST_COEF_YEARS[1]))
+  # aggregate the D_driver by region and land type
+    L125.bcoc_tgbkm2_R_defor_2000 <-  L125.bcoc_tgbkm2_R_GLU_defor_2000 %>%
+      group_by(GCAM_region_ID, Land_Type) %>%
+      summarize_at(vars(D_driver),sum)
+  # calculate deforestation and forest fire emission factors
```

Comment:
I was thinking something like 
```R
L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj %>% 
    arrange(year) %>%
    filter( year %in% emissions.DEFOREST_COEF_YEARS) %>%
    group_by(GCAM_region_ID, Land_Type) %>% 
    mutate(D_driver = pmax(value[1] - value[2], 0)/ (emissions.DEFOREST_COEF_YEARS[2] - emissions.DEFOREST_COEF_YEARS[1])) %>% 
    group_by(GCAM_region_ID, Land_Type) %>% 
    summarise(D_driver = sum(D_driver))
```

(or something like that, not sure) ... but probably not worth the trouble to change.